samples/ordered_puzzle/components/ordered_puzzle.js

```diff
@@ -0,0 +1,94 @@
+AFRAME.registerComponent("ordered-puzzle", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+
+  multiple: true,
+
+  init: function () {
+    const data = this.data.jsonData;
+    const el = this.el;
+
+    const numPuzzlePieces = data.images.length;
+
+    for (i = 0; i < numPuzzlePieces; i++) {
+      var textBoxProp =
+        '{"width": "' +
+        data.images[i].width +
+        '", "height": "' +
+        data.images[i].height +
+        '", "color": "yellow", "x": "' +
+        data.images[i].xTarget +
+        '", "y": "' +
+        data.images[i].yTarget +
+        '", "z": "0", "text": "' +
+        i +
+        '"}';
+
+      this.target = document.createElement("a-entity");
+      this.target.setAttribute("id", "puzzle-target");
+      this.target.setAttribute("target-box", "jsonData", textBoxProp);
+      this.el.appendChild(this.target);
+
+      this.puzzlePiece = document.createElement("a-image");
+      this.puzzlePiece.setAttribute("id", "puzzle-piece-image");
+      this.puzzlePiece.setAttribute("src", data.images[i].imageSrc);
+      this.puzzlePiece.setAttribute("width", data.images[i].width);
+      this.puzzlePiece.setAttribute("height", data.images[i].height);
+      this.puzzlePiece.setAttribute(
+        "position",
+        data.images[i].x + " " + data.images[i].y + " 0"
+      );
+      this.puzzlePiece.setAttribute("xTarget", data.images[i].xTarget);
+      this.puzzlePiece.setAttribute("yTarget", data.images[i].yTarget);
+      this.puzzlePiece.setAttribute("onTarget", false);
+      this.puzzlePiece.setAttribute("class", "draggable");
+      this.puzzlePiece.addEventListener("dragend", onDragEnd);
+      this.el.appendChild(this.puzzlePiece);
+    }
+  },
+});
+
+function onDragEnd(event) {
+  event.target.setAttribute("onTarget", isOnTarget(event.target));
+  if (isPuzzleComplete()) {
+    closePuzzle();
+  }
+}
+
+function isOnTarget(puzzlePiece) {
+  const delta = puzzlePiece.object3D.children[0].position;
+  const originalPos = puzzlePiece.getAttribute("position");
```

Comment:
I am a little bit confused here... Why are the two positions different? Like puzzlePiece.object3D.children[0].position vs puzzlePiece.getAttribute("position"), do you mind explaining what each of them represent?

Reply:
I originally thought that these two positions were the same as well, but when I was testing it out it turns out the puzzlePiece.object3D.children[0].position represents the position of the object relative to the initial position that was set by puzzlePiece.getAttribute("position").

Reply:
Ah I see, that's interesting. I never knew that! Thanks for the explanation.